requester/nsq_requester.go

```diff
@@ -0,0 +1,84 @@
+package requester
+
+import (
+	"strconv"
+
+	"github.com/nsqio/go-nsq"
+	"github.com/tylertreat/bench"
+)
+
+// NSQRequesterFactory implements RequesterFactory by creating a Requester
+// which publishes messages to NSQ and waits to receive them.
+type NSQRequesterFactory struct {
+	URL         string
+	PayloadSize int
+	Topic     string
+}
+
+// GetRequester returns a new Requester, called for each Benchmark connection.
+func (n *NSQRequesterFactory) GetRequester(num uint64) bench.Requester {
+	return &nsqRequester{
+		url:         n.URL,
+		payloadSize: n.PayloadSize,
+		topic:     n.Topic + "-" + strconv.FormatUint(num, 10),
+		channel:   n.Topic + "-" + strconv.FormatUint(num, 10),
+	}
+}
+
+// nsqRequester implements Requester by publishing a message to NSQ and
+// waiting to receive it.
+type nsqRequester struct {
+	url         string
+	payloadSize int
+	topic     string
+	channel   string
+	producer  *nsq.Producer
+	consumer  *nsq.Consumer
+	msg			[]byte
+}
+
+// Setup prepares the Requester for benchmarking.
+func (n *nsqRequester) Setup() error {
+	config := nsq.NewConfig()
+	producer, err := nsq.NewProducer(n.url, config)
+	if err != nil {
+		return err
+	}
+	consumer, err := nsq.NewConsumer(n.topic, n.channel, config)
+	if err != nil {
+		return err
+	}
+	consumer.AddConcurrentHandlers(nsq.HandlerFunc(func(m *nsq.Message) error {
+		return nil
+	}),1)
+	if err := consumer.ConnectToNSQD(n.url); err != nil {
+		producer.Stop()
+		return err
+	}
+
+	n.producer = producer
+	n.consumer = consumer
+	n.msg = make([]byte, n.payloadSize)
+
+	return nil
+}
+
+// Request performs a synchronous request to the system under test.
+func (n *nsqRequester) Request() error {
+	if err := n.producer.Publish(n.topic, n.msg); err != nil {
+		return err
+	}
+	return nil
```

Comment:
My point is other messaging-based `Requester`s in this package are measuring the time from when a message is published to when it's received on the other end. For example, see the [Kafka one](https://github.com/tylertreat/bench/blob/master/requester/kafka_requester.go#L79-L85). 

It's perfectly fine to just want to measure the latency of publish calls, but it might be disingenuous to users if they are comparing with other systems since they would be comparing end-to-end latency with publish latency. If we are only interested in the latency of publishes for nsq, perhaps this should be named `NSQPublishRequester` or something along those lines? Or, at the very least, just clearly document this in a docstring.